catalog/core/catalog-core-solrcommands/src/test/java/org/codice/ddf/commands/solr/TestBackupCommand.java

```diff
@@ -142,6 +199,296 @@ public void testSystemPropertiesNotSet() throws Exception {
+            SolrClient getSolrClient() {
+                return miniSolrCloud.getSolrClient();
+            }
+
+            // We get the solr client from the MiniSolrCloudCluster, so we don't
+            // want to shut it down after each test. We don't create a MiniSolrCloudCluster
+            // for each test to reduce the time it takes to run the tests.
+            @Override
+            void shutdown(SolrClient client) {}
+        };
+        backupCommand.backupLocation = getBackupLocation();
+        backupCommand.coreName = INVALID_COLLECTION_NAME;
+
+        // Perform Test
+        backupCommand.doExecute();
+
+        // Verify
+        assertThat(consoleOutput.getOutput(),
+                containsString(String.format(
+                        "Backing up collection [%s] to shared location [%s] using backup name [%s_",
+                        INVALID_COLLECTION_NAME,
+                        backupCommand.backupLocation,
+                        INVALID_COLLECTION_NAME)));
+        assertThat(consoleOutput.getOutput(),
+                containsString(String.format("Backup failed. Unable to optimize collection [%s]. Collection not found: %s",
+                        INVALID_COLLECTION_NAME,
+                        INVALID_COLLECTION_NAME)));
+    }
+
+    @Test
+    public void testPerformSolrCloudAsynchronousBackupAndStatus() throws Exception {
+
+        // Set the solr client type system property so that the
+        // BackupCommand knows that it needs to backup solr cloud.
+        setupSolrClientType(SOLR_CLOUD_CLIENT_TYPE);
+
+        // Setup BackupCommand
+        BackupCommand backupCommand1 = new BackupCommand() {
+            @Override
+            SolrClient getSolrClient() {
+                return miniSolrCloud.getSolrClient();
+            }
+
+            // We get the solr client from the MiniSolrCloudCluster, so we don't
+            // want to shut it down after each test. We don't create a MiniSolrCloudCluster
+            // for each test to reduce the time it takes to run the tests.
+            @Override
+            void shutdown(SolrClient client) {}
+        };
+        backupCommand1.asyncBackup = true;
+        backupCommand1.backupLocation = getBackupLocation();
+        backupCommand1.coreName = DEFAULT_CORE_NAME;
+
+        // Perform Test
+        backupCommand1.doExecute();
+
+        // Verify
+        assertThat(consoleOutput.getOutput(),
+                containsString(String.format(
+                        "Backing up collection [%s] to shared location [%s] using backup name [%s_",
+                        DEFAULT_CORE_NAME,
+                        backupCommand1.backupLocation,
+                        DEFAULT_CORE_NAME)));
+        assertThat(consoleOutput.getOutput(), containsString("Solr Cloud backup request Id:"));
+
+        String requestId = getRequestId(consoleOutput.getOutput());
+        consoleOutput.reset();
+
+        BackupCommand backupCommand2 = new BackupCommand() {
```

Comment:
You could remove the numbers on these backupCommand names and name this one "backupStatusCommand" or something similar.